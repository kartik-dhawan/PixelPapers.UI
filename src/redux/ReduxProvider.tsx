"use client"

/**
 *
 * This component acts as a parent for client side components that require redux store
 * Wrap the component with this 'Redux Provider' to use state management on client side
 *
 * the updates made on the client side in the store will not be reflected in the server side components
 *
 * SERVER SIDE:
 * - Dispatching: store.dispatch(functionName(payload))
 * - Getting data from store: store.getState().sliaceName.stateName
 *
 * CLIENT SIDE:
 * - Dispatching: dispatch(functionName(payload)) [useDispatch()]
 * - Getting data from store: useSelector((state)=>state.slice.statename)
 *
 * Ref: https://www.youtube.com/watch?v=dRLjoT4r-jc
 */

import { Provider } from "react-redux"
import { store } from "./store"
import { ChildrenProps } from "@/utils/interfaces"

const ReduxProvider = ({ children }: ChildrenProps) => {
  return <Provider store={store}>{children}</Provider>
}

export default ReduxProvider
