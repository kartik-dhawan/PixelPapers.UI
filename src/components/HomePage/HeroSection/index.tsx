"use client"

import { RootType } from "@/redux/store"
import { anton, poppins } from "@/utils/fonts"
import { Box, Button, Stack, SvgIcon } from "@mui/material"
import Link from "next/link"
import { useSelector } from "react-redux"
import Image from "next/image"
import { styles } from "./styles"
import { useState } from "react"
import TrendingFlatIcon from "@mui/icons-material/TrendingFlat"

const HeroSection = () => {
  const { content } = useSelector((state: RootType) => state.contentSlice)
  const [isImageLoaded, setIsImageLoaded] = useState<boolean>(false)

  return (
    <Box component="section" sx={{ height: "100svh", position: "relative" }}>
      <Box sx={styles.heroSectionImageWrapper}>
        {!isImageLoaded && <Box sx={styles.heroSectionImageFallback} />}
        <Image
          onLoad={() => {
            setIsImageLoaded(true)
          }}
          src={`https:${content?.homePageImage?.fields.file.url}`}
          alt={content.homePageImage?.fields.description ?? "Image"}
          fill
          sizes="(max-width: 768px) 100vw, (max-width: 1200px) 50vw, 50vw)"
        />
      </Box>
      <Stack
        justifyContent="center"
        alignItems="center"
        sx={styles.heroSectionOverlayContentWrapper}
      >
        <Box
          component="h1"
          className={anton.className}
          sx={styles.heroSectionHeading}
        >
          PIXEL PAPERS
        </Box>
        <Box className={poppins.className} sx={styles.heroSectionAuthorLink}>
          by{" "}
          <Link
            href={content.curatedByLink ?? "#!"}
            target="_blank"
            rel="noreferrer"
            aria-label="View the instagram profile of Kartik Dhawan (The curator of this website)"
          >
            kartik-dhawan
          </Link>
        </Box>
      </Stack>
      <Button
        className={poppins.className}
        disableRipple
        sx={styles.heroSectionButton}
        endIcon={
          <SvgIcon className="hello" sx={styles.heroSectionStartReadingSvg}>
            <TrendingFlatIcon />
          </SvgIcon>
        }
      >
        <Link href="/travel">Start Reading</Link>
      </Button>
    </Box>
  )
}

export default HeroSection
