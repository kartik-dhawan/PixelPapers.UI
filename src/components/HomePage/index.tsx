import React from "react"
import HeroSection from "./HeroSection"

const HomePage = () => {
  return (
    <React.Fragment>
      {/* HERO SECTION */}
      <HeroSection />
    </React.Fragment>
  )
}

export default HomePage
