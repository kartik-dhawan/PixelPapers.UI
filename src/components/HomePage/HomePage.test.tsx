import ReduxProvider from "@/redux/ReduxProvider"
import "@testing-library/jest-dom"
import renderer from "react-test-renderer"
import HomePage from "."

describe("Home page & all its components", () => {
  it("-- should match the snapshot", () => {
    const tree = renderer.create(
      <ReduxProvider>
        <HomePage />
      </ReduxProvider>,
    )

    expect(tree).toMatchSnapshot()
  })
})
