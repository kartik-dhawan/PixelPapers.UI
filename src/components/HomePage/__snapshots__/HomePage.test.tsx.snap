// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page & all its components -- should match the snapshot 1`] = `
<section
  className="MuiBox-root css-n93qtu"
>
  <div
    className="MuiBox-root css-653cb7"
  >
    <div
      className="MuiBox-root css-wf3u97"
    />
    <img
      alt="Image"
      data-nimg="fill"
      decoding="async"
      fetchpriority="high"
      onError={[Function]}
      onLoad={[Function]}
      sizes="(max-width: 768px) 100vw, (max-width: 1200px) 50vw, 50vw)"
      src="/_next/image?url=https%3Aundefined&w=3840&q=75"
      srcSet="/_next/image?url=https%3Aundefined&w=384&q=75 384w, /_next/image?url=https%3Aundefined&w=640&q=75 640w, /_next/image?url=https%3Aundefined&w=750&q=75 750w, /_next/image?url=https%3Aundefined&w=828&q=75 828w, /_next/image?url=https%3Aundefined&w=1080&q=75 1080w, /_next/image?url=https%3Aundefined&w=1200&q=75 1200w, /_next/image?url=https%3Aundefined&w=1920&q=75 1920w, /_next/image?url=https%3Aundefined&w=2048&q=75 2048w, /_next/image?url=https%3Aundefined&w=3840&q=75 3840w"
      style={
        {
          "bottom": 0,
          "color": "transparent",
          "height": "100%",
          "left": 0,
          "objectFit": undefined,
          "objectPosition": undefined,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": "100%",
        }
      }
    />
  </div>
  <div
    className="MuiStack-root css-1v9ayl4-MuiStack-root"
  >
    <h1
      className="className MuiBox-root css-144oidx"
    >
      PIXEL PAPERS
    </h1>
    <div
      className="className MuiBox-root css-10qd5jz"
    >
      by
       
      <a
        aria-label="View the instagram profile of Kartik Dhawan (The curator of this website)"
        href="#!"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
        rel="noreferrer"
        target="_blank"
      >
        kartik-dhawan
      </a>
    </div>
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium className css-fivqy-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <a
      href="/travel"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      Start Reading
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium hello css-1yvb69x-MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="TrendingFlatIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m22 12-4-4v3H3v2h15v3z"
          />
        </svg>
      </svg>
    </a>
  </button>
</section>
`;
